### PR TITLE
4.9.2 - v4 Migration Cleanup

### PR DESCRIPTION
**Pull request by brainlulz**
This PR covers issue #85 .

- [ ] This PR handles the issue and requires no additional PRs.
- [x] You have validated the need for this change.

**What did this PR accomplish?**

- Fix linting
- Remove wikilinks

Thank you for your contribution!

### CODE REVIEW

File: document/4_Web_Application_Security_Testing/4.9_Testing_for_Error_Handling/4.9.2_Testing_for_Stack_Traces_OTG-ERR-002.md [hunk 1]
```diff
@@ -1,47 +1,46 @@
...
+- input that contains non alphanumeric characters or query syntax.
+- empty inputs.
+- inputs that are too long.
+- access to internal pages without authentication.
+- bypassing application flow.
 
 All the above tests could lead to application errors that may contain stack traces. It is recommended to use a fuzzer in addition to any manual testing.
 
-Some tools, such as [OWASP ZAP](OWASP_Zed_Attack_Proxy_Project "wikilink") and Burp proxy will automatically detect these exceptions in the response stream as you are doing other penetration and testing work.
+Some tools, such as [OWASP ZAP](https://www.owasp.org/index.php/OWASP_Zed_Attack_Proxy_Project) and Burp proxy will automatically detect these exceptions in the response stream as you are doing other penetration and testing work.
```
Comment (ThunderSon) [inline document/4_Web_Application_Security_Testing/4.9_Testing_for_Error_Handling/4.9.2_Testing_for_Stack_Traces_OTG-ERR-002.md]: Add a link to burp proxy

File: document/4_Web_Application_Security_Testing/4.9_Testing_for_Error_Handling/4.9.2_Testing_for_Stack_Traces_OTG-ERR-002.md [hunk 2]
```diff
@@ -1,47 +1,46 @@
...
 
 ### Gray Box Testing
 
 Search the code for the calls that cause an exception to be rendered to a String or output stream. For example, in Java this might be code in a JSP that looks like:
 
-    &lt;% e.printStackTrace( new PrintWriter( out ) ) %&gt;
+```Java
+   &lt;% e.printStackTrace( new PrintWriter( out ) ) %&gt;
```
Comment (ThunderSon) [inline document/4_Web_Application_Security_Testing/4.9_Testing_for_Error_Handling/4.9.2_Testing_for_Stack_Traces_OTG-ERR-002.md]: Replace the html encoding with normal characters `<` and `>`

File: document/4_Web_Application_Security_Testing/4.9_Testing_for_Error_Handling/4.9.2_Testing_for_Stack_Traces_OTG-ERR-002.md [hunk 3]
```diff
@@ -1,47 +1,46 @@
...
 
-Tools
------
-
-\[1\] ZAP Proxy - https://www.owasp.org/index.php/OWASP\_Zed\_Attack\_Proxy\_Project 
+## Tools
 
-References
-----------
+- [ZAP Proxy](https://www.owasp.org/index.php/OWASP_Zed_Attack_Proxy_Project)
```
Comment (ThunderSon) [inline document/4_Web_Application_Security_Testing/4.9_Testing_for_Error_Handling/4.9.2_Testing_for_Stack_Traces_OTG-ERR-002.md]: Add the burp proxy to this list as well

File: document/4_Web_Application_Security_Testing/4.9_Testing_for_Error_Handling/4.9.2_Testing_for_Stack_Traces_OTG-ERR-002.md [hunk 4]
```diff
@@ -1,47 +1,48 @@
...
 
 Stack traces are not vulnerabilities by themselves, but they often reveal information that is interesting to an attacker. Attackers attempt to generate these stack traces by tampering with the input to the web application with malformed HTTP requests and other input data.
 
 If the application responds with stack traces that are not managed it could reveal information useful to attackers. This information could then be used in further attacks. Providing debugging information as a result of operations that generate errors is considered a bad practice due to multiple reasons. For example, it may contain information on internal workings of the application such as relative paths of the point where the application is installed or how objects are referenced internally.
 
-How to Test
------------
+## How to Test
 
 ### Black Box testing
```
Comment (kingthorin) [inline document/4_Web_Application_Security_Testing/4.9_Testing_for_Error_Handling/4.9.2_Testing_for_Stack_Traces_OTG-ERR-002.md]: Could you make all headings use title caps?

https://github.com/OWASP/OWASP-Testing-Guide-v5#style-guidelines

File: document/4_Web_Application_Security_Testing/4.9_Testing_for_Error_Handling/4.9.2_Testing_for_Stack_Traces_OTG-ERR-002.md (hunk 3 above)
Comment (kingthorin) [reply document/4_Web_Application_Security_Testing/4.9_Testing_for_Error_Handling/4.9.2_Testing_for_Stack_Traces_OTG-ERR-002.md]: You did add it to the earlier content but not here. I'm not sure we need them listed both places, but whatever happens I guess it should be balanced.

**Pull request merged**